src/Util/ConfigUtil.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * @file
+ * Contains \Gush\Util\ConfigUtil.
+ */
+
+namespace Gush\Util;
+
+class ConfigUtil {
+
+  public static function generateConfigurationIdentifier($adapter, array $adapterConfig) {
+    return $adapter . ':' . substr(sha1(serialize($adapterConfig)), 0, 8);
+  }
+
```

Comment:
CS issues are addressed in my branch, just need some time to change some things to not introduce the many BC breaks ;)